src/holofuel/pages/Home/Home.js

```diff
@@ -45,7 +51,16 @@ export default function Home () {
   const history = useHistory()
   const goToOfferRequest = () => history.push(OFFER_REQUEST_PATH)
 
-  const { data: { holofuelLedger: { balance: holofuelBalance } = { balance: 0 } } = {} } = useQuery(HolofuelLedgerQuery, { fetchPolicy: 'network-only' })
+  const { newMessage } = useFlashMessageContext()
+  const [errorMessage, setErrorMessage] = useState(null)
+
+  if (errorMessage) {
+    newMessage(errorMessage)
+    setErrorMessage('')
+  }
+
+  const filterActionableTransactionsByStatus = status => holofuelActionableTransactions.filter(actionableTx => actionableTx.status === status)
+  if (!isEmpty(filterActionableTransactionsByStatus('declined')) && errorMessage === null) setErrorMessage(declinedTransactionNotice)
```

Comment:
`newMessage` is a side effect, so you should not make calls to it directly in the render function (on line 58) (no side effects in the render function outside of hooks).
Same with `setErrorMessage` on lines 59 and 63.

you don't actually need state here, what you want is `useEffect`

```
useEffect(() => {
  if (!isEmpty(filterActionableTransactionsByStatus('declined')) && errorMessage === null) {
    newMessage(declinedTransactionNotice)
  } 
}, [filterActionableTransactionsByStatus])
```

So that's how you would achieve this functionality, BUT I'm this seems like bad UX to me, so let's chat and make sure this is how we want to handle this situation, and potentially bring in C and/or Z to think about it.

Reply:
Yes, this was a placeholder message for a larger conversation.  NB: Had the conversation with Z and Rob yesterday, this message will become a modal with a bulk action associated with it.  I will leave this placeholder for now.. as this will be the foundation upon which we can use for the updates.